perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ConsentsManagerBlImpl.java

```diff
@@ -72,6 +193,23 @@ public ConsentHub createConsentHub(PerunSession sess, ConsentHub consentHub) thr
 
 	@Override
 	public void deleteConsentHub(PerunSession sess, ConsentHub consentHub) throws ConsentHubAlreadyRemovedException {
+
+		// Check if ConsentHub exists
+		try {
+			getConsentHubById(sess, consentHub.getId());
```

Comment:
I think we can omit this check because this method in the impl layer would throw this exception anyway. What do you think?

Reply:
Okay, you can delete this then.